### ISSUE TITLE
Repeatable group field issue:  row classes not updated dynamically.

### ISSUE THREAD

**Issue by thirumani02** [WebDevStudios/CMB2#689]
Hello,
I'm trying to show/hide the fields based on the value of other field on my repeatable group fields so i need to check the conditions based on the group id & field id dynamically.

Here is an issue, since the row classes are updated only after saving/publishing the page, i can't show/hide the fields dynamically.

This is what i see on firebug initally.

    <div class="postbox cmb-row cmb-repeatable-grouping" data-iterator="0">
        <button>...</button>
        <div class="cmbhandle">Click to toggle</div>
        <h3 class="cmb-group-title cmbhandle-title">Entry 1</h3>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id0]-0-[field_id0] cmb-repeat-group-field"></div>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id1]-0-[field_id1] cmb-repeat-group-field"></div>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id2]-0-[field_id2] cmb-repeat-group-field"></div>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id3]-0-[field_id3] cmb-repeat-group-field"></div>
    </div>

Once i clicked the "Add New Entry" my data-iterator get updated dynamically but my repeatable fields didn't. Actually, it should be like this cmb2-id-[group_id0]-**1**-[field_id0], cmb2-id-[group_id1]-**1**-[field_id1], cmb2-id-[group_id2]-**1**-[field_id2], etc.

    <div class="postbox cmb-row cmb-repeatable-grouping" data-iterator="0">
        <button>...</button>
        <div class="cmbhandle">Click to toggle</div>
        <h3 class="cmb-group-title cmbhandle-title">Entry 1</h3>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id0]-0-[field_id0] cmb-repeat-group-field"></div>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id1]-0-[field_id1] cmb-repeat-group-field"></div>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id2]-0-[field_id2] cmb-repeat-group-field"></div>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id3]-0-[field_id3] cmb-repeat-group-field"></div>
    </div>
    <div class="postbox cmb-row cmb-repeatable-grouping" data-iterator="1">
        <button>...</button>
        <div class="cmbhandle">Click to toggle</div>
        <h3 class="cmb-group-title cmbhandle-title">Entry 2</h3>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id0]-0-[field_id0] cmb-repeat-group-field"></div>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id1]-0-[field_id1] cmb-repeat-group-field"></div>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id2]-0-[field_id2] cmb-repeat-group-field"></div>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id3]-0-[field_id3] cmb-repeat-group-field"></div>
    </div>

After clicking Save/publish, its perfect and all my conditions are working.

    <div class="postbox cmb-row cmb-repeatable-grouping" data-iterator="0">
        <button>...</button>
        <div class="cmbhandle">Click to toggle</div>
        <h3 class="cmb-group-title cmbhandle-title">Entry 1</h3>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id0]-0-[field_id0] cmb-repeat-group-field"></div>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id1]-0-[field_id1] cmb-repeat-group-field"></div>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id2]-0-[field_id2] cmb-repeat-group-field"></div>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id3]-0-[field_id3] cmb-repeat-group-field"></div>
    </div>
    <div class="postbox cmb-row cmb-repeatable-grouping" data-iterator="1">
        <button>...</button>
        <div class="cmbhandle">Click to toggle</div>
        <h3 class="cmb-group-title cmbhandle-title">Entry 2</h3>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id0]-1-[field_id0] cmb-repeat-group-field"></div>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id1]-1-[field_id1] cmb-repeat-group-field"></div>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id2]-1-[field_id2] cmb-repeat-group-field"></div>
        <div class="cmb-row cmb-type-file cmb2-id-[group_id3]-1-[field_id3] cmb-repeat-group-field"></div>
    </div>

Awaiting for your response :)

**Comment by jtsternberg** [comment]
You might try checking out this extension: https://github.com/jcchavezs/cmb2-conditionals

**Comment by thirumani02** [comment]
Yes, I'm asking for this extension. This extension is not working on repeatable fields, i can give you a live link and login credentials(privately- thirumani02@gmail.com) if necessary. 

Thank you

**Comment by jtsternberg** [comment]
Ok, then you will need to open an issue on that repository, as that is not ours.

**Comment by thirumani02** [comment]
Think is, it may be because of the wrong ID representation by CMB2. They're trying to target the field based on the repeatable group ID. But CMB2 update the ID(on the particular element) only after saving the page.

Btw: thanks for your time. I will raise the ticket on that repository too.

**Comment by tw2113** [comment]
Any feedback from the cmb2-conditionals devs @thirumani02 ?

**Comment by thirumani02** [comment]
@tw2113 I just found a thread with same issue https://github.com/jcchavezs/cmb2-conditionals/issues/2 and they labeled as an enhancement.

**Comment by ShariqKhan2012** [comment]
I am facing the same problem. Does anyone has an idea how to solve this?
